packages/taco/test/conditions/context.test.ts

```diff
@@ -117,10 +119,11 @@ describe('context', () => {
     });
 
     it('detects if a signer is required', () => {
-      const conditionObj = {
+      const conditionObj: ContractConditionProps = {
         ...testContractConditionObj,
         returnValueTest: {
-          ...testReturnValueTest,
+          index: 0,
```

Comment:
`testContractConditionObj` is an ERC20 `balanceOf` contract call that returns a `uint256`,`index` is not applicable.

```suggestion
```

Also is this the best test - it is technically an invalid condition since `USER_ADDRESS_PARAM` is not valid as a return value type since the contract returns a `uint256`. You could instead do an NFT `ownerOf` condition.

Reply:
Resolved other comments but this comment about the invalidity of this test still stands - the condition is invalid. 

Alternatively, you could set it up that the parameters used for `testContractConditionObj` use `:userAddress` context variable instead of a specific value (which is does currently). But the returnValueTest `value` to compare is for an int, not a user address.

Reply:
So there are comparators that don't make sense for certain parameter types? I would think we can '>' compare strings since strings have an (alphanumeric) ordering. But we can also constrain the type of comparator on the parameter type.

Reply:
It's not the parameter types, but there are return values that don't make sense based on what the contract function is expected to return. The `testContractConditionObj`'s method is ERC20.balanceOf which is supposed to return a `uint256`. However, the returnValueTest is being modify to use an `address` as the comparator value. So the condition is attempting to compare a `unit256` to an `address` which is confusing.